### PR TITLE
api: python: checkSalt() implemented #539

### PR DESCRIPTION
**Pull request by AATDev21**
- Added new IndigoObject method checkSalt with a list of basic ions for search.
- Added unittest for this method.
- Fixed typos in structure_checker.cpp, test.js and ref basic_check.py.out.

### CODE REVIEW

**Review by MysterionRise — CHANGES_REQUESTED** [review]
@AATDev21  why we need basic_check.py.out file in Git?


Could you also run black + isort + pylint on the fresh additions? It looks like some of the PEP-8 guidelines are missing here

File: api/python/tests/test_indigo.py [hunk 1]
```diff
@@ -9,3 +9,7 @@ def test_aromatize_smiles(self) -> None:
         m = self.indigo.loadMolecule('C1=CC=CC=C1')
         m.aromatize()
         self.assertEqual('c1ccccc1', m.smiles())
+
+    def test_check_salt(self) -> None:
+        m = self.indigo.loadMolecule('[Na+].C')
+        self.assertTrue(m.checkSalt())
```
Comment (mkviatkovskii) [inline api/python/tests/test_indigo.py:15]: Let's add a couple more tests:
1. To return `True` if compound contains multiple salts.
2. To return `False` if compound does not contain any salts.
3. To return `False` if compound contains, some salt-like ion, but not as standalone fragment.

**Review by mkviatkovskii — CHANGES_REQUESTED** [review]
Looks good, let's add a couple more tests and the merge.

**Comment by mkviatkovskii** [comment]
@MysterionRise, we keep *.out file as a reference in our own-written mind-blowing test engine. Not very intuitive, but it's how it works for now.

**Comment by MysterionRise** [comment]
@mkviatkovskii that's weird, but okay :)

File: api/python/tests/test_indigo.py (hunk 1 above)
Comment (AATDev21) [reply api/python/tests/test_indigo.py:15]: I'll add them, thank you!

**Comment by AATDev21** [comment]
@MysterionRise , I didn't change or add any imports in here, but I'll check. Thank you!

**Comment by MysterionRise** [comment]
@AATDev21 im more on the side of 2 lines between methods, etc

**Comment by AATDev21** [comment]
@MysterionRise I just found a link [https://www.python.org/dev/peps/pep-0008/#blank-lines](url) with PEP8, in which one blank line is recommended for class methods separation. I could use extra lines if my method was entirely different from others in the class, but it just continued a large group of already existed molecule methods.

**Review by mkviatkovskii — APPROVED** [review]

**Pull request merged**